lib/index.js

```diff
@@ -359,80 +217,111 @@ exports.createFunction = async (ctx, functionSpecification) => {
-        expiresIn: expiry,
-        audience: ctx.configuration.fusebit_storage_audience,
-        issuer: ctx.configuration.fusebit_storage_issuer_id,
-        subject: ctx.configuration.fusebit_storage_subject,
-        keyid: ctx.configuration.fusebit_storage_key_id,
-        header: { jwtId: Date.now().toString() },
-    });
-
-    const url = generateStorageUrl(ctx.configuration);
+const removeLeadingSlash = (s) => s.replace(/^\/(.+)$/, '$1');
+const removeTrailingSlash = (s) => s.replace(/^(.+)\/$/, '$1');
+
+exports.createStorageClient = async (ctx, accessToken, storageIdPrefix) => {
+    storageIdPrefix = storageIdPrefix ? removeLeadingSlash(removeTrailingSlash(storageIdPrefix)) : '';
+    const functionUrl = Url.parse(ctx.baseUrl);
+    let storageBaseUrl = `${
+            functionUrl.protocol
+        }//${
+            functionUrl.host
+        }/v1/account/${
+            ctx.accountId
+        }/subscription/${
+            ctx.subscriptionId
+        }/storage${
+            storageIdPrefix ? '/' + storageIdPrefix : ''
+        }`;
+
+    const getUrl = (storageSubId) => {
+        storageSubId = storageSubId ? removeTrailingSlash(removeLeadingSlash(storageSubId)) : '';
+        return `${storageBaseUrl}${storageSubId ? ('/' + storageSubId) : ''}`;
+    }
 
-    return {
-        etag: null,
-        expiration: Date.now() + expiry * 1000,
-        get: async function () {
-            const response = await Superagent.get(url)
+    const storageClient = {
+        get: async function (storageSubId) {
+            const response = await Superagent.get(getUrl(storageSubId))
                 .set('Authorization', `Bearer ${accessToken}`)
                 .ok((res) => res.status < 300 || res.status === 404);
-            this.etag = response.body.etag;
-            return response.status === 404 ? undefined : response.body.data;
+            return response.status === 404 ? undefined : response.body;
         },
-        put: async function (data, force) {
-            let request = Superagent.put(url).set('Authorization', `Bearer ${accessToken}`);
-            let payload = { data: data };
-            if (!force) {
-                payload.etag = this.etag;
+        put: async function (data, storageSubId) {
+            const response = await Superagent.put(getUrl(storageSubId))
+                .set('Authorization', `Bearer ${accessToken}`)
+                .send(data);
+            return response.body;
+        },
+        delete: async function (storageSubId, recursive, forceRecursive) {
+            storageSubId = storageSubId ? removeLeadingSlash(removeTrailingSlash(storageSubId)) : '';
+            if (!storageSubId && !storageIdPrefix && recursive && !forceRecursive) {
+                throw new Error('You are attempting to recursively delete all storage objects in the Fusebit subscription. If this is your intent, please pass "true" as the third parameter in the call to delete(storageSubId, recursive, forceRecursive).');
```

Comment:
Scary.  Seems like it should be enforced serverside if we're going to enforce it anywhere...  no change needed, but it might be worth doing.